src/aotcompile.cpp

```diff
@@ -267,7 +267,6 @@ void *jl_create_native_impl(jl_array_t *methods, LLVMOrcThreadSafeModuleRef llvm
     jl_method_instance_t *mi = NULL;
     jl_code_info_t *src = NULL;
     JL_GC_PUSH1(&src);
-    JL_LOCK(&jl_codegen_lock);
```

Comment:
You have inverted the ordering of `jl_codegen_lock` and the `ThreadSafeModule` lock. Can you update the devlocks to reflect that? Is that even valid?

Reply:
So the lock is taken in small spurts to set up the TSM, but we know that doesn't take the codegen lock, and we release the context lock before attempting to take the codegen lock, and then reacquire the context lock during the params constructor. Since the context lock isn't actively being held during the acquisition of the codegen lock, there should be no priority inversion occurring.

Reply:
Ah, okay, I assumed `clone.getContext` was getting the lock, not `jl_codegen_params_t`